File: CS50x Problemsets/2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int get_wordcount(string text);
double get_letters(string text, int wordcount);
double get_sentences(string text, int wordcount);
int coleman_liau(double l, double s);

/* L= avrg num of letters, not chars per 100 words
    s= num of scentences per 100 words
        index = 0.0588 * L - 0296 * S - 15.8*/
int main(void)
{
    string text = get_string("Text: ");
    int wordcount = get_wordcount(text);
    double letters_per_100 = get_letters(text, wordcount);
    double senctencves_per_100 = get_sentences(text, wordcount);
    int index = coleman_liau(letters_per_100, senctencves_per_100);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}
/*
< 1before grade 1
>16 grade 16+
*/
int coleman_liau(double l, double s)
{
    double index = 0.0588 * l - 0.296 * s - 15.8;
    int index2 = round(index);
    return index2;
}

int get_wordcount(string text)
{
    int wordcount = 1;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isspace(text[i]) != 0)
            wordcount++;
    }
    // printf("wordcount: %i\n", wordcount);
    return wordcount;
}

double get_letters(string text, int wordcount)
{
    double lettercount = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]) != 0)
        {
            lettercount++;
        }
    }
    // printf("lettercount: %f\n", lettercount);
    lettercount = lettercount / wordcount * 100;
    return lettercount;
}

double get_sentences(string text, int wordcount)
{
    double sentencecount = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (strchr("?.!", text[i]) != NULL)
        {
            sentencecount++;
        }
    }
    sentencecount = sentencecount / wordcount * 100;
    // printf("sentencecount: %i\n", sentencecount);
    return sentencecount;
}

File: CS50x Problemsets/6/readability_py/readability.py
""" L= avrg num of letters, not chars, per 100 words
    s= num of scentences per 100 words
        index = 0.0588 * L - 0296 * S - 15.8 """


def main():
    text = input("Text: ")
    words = get_words(text)
    letters = get_letters(text, words)
    sentences = get_sentences(text, words)
    index = round((0.0588 * letters - 0.296 * sentences - 15.8))
    # print(letters)

    # print grade according to index
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


# get word count to calculate other meassures per 100 words
def get_words(text):
    # set to 1 since there is no space after the last word
    words = 1
    for i in range(len(text)):
        if text[i].isspace() == True:
            words += 1
    return words


# getting the amount of letters per 100 words in text
def get_letters(text, words):
    # initialise letter counter
    letters = 0
    # going thru every letter and count if alphanumeric
    for i in range(len(text)):
        if text[i].isalpha() == True:
            letters += 1
    # letters per word times 100
    letters = letters / words * 100
    return letters


# get number of sentences per 100 words
def get_sentences(text, words):
    # set counter
    sentences = 0
    # count every time a scentence ends with . ? !
    for i in range(len(text)):
        if text[i] in ".?!":
            sentences += 1
    # sentences per words times 100
    sentences = sentences / words * 100
    return sentences


if __name__ == "__main__":
    main()
